src/python/test/test_extract_expression.py

```diff
@@ -78,6 +102,30 @@ def standard_positive_filter_test(self, json_name):
 
         validator.validate(input_json)
 
+    #
+    # Path Validation tests
+    # There are # ways this function can fail.  Each of the following assumes the previous conditions have been met:
+    # 1. Object in wrong project
+    # 2. Object not of class record
+    # 3. Object not of recordType "Dataset" or "CohortBrowser"
+    # 4. Object is not of correct version (3.0 at the time of this writing)
+    # 5. Object is CohortBrowser type and --assay-name or --list-assays has been given on the command line
+    #
+
+    # 2. Object not of class record
+    def test_bad_dataset_type_unit(self):
+        test_record = self.wrong_type_path_file
+        expected_error_message = "{}: Invalid path. The path must point to a record type of cohort or dataset".format(test_record)
+        self.standard_path_validation_test(test_record,expected_error_message)
+
+
+    # 3. Object not of recordType "Dataset" or "CohortBrowser"
```

Comment:
This comment should point to `4. Object is not of correct version (3.0 at the time of this writing)` instead, right?

Reply:
yep, updated